billing/data/consumer.js

```diff
@@ -0,0 +1,83 @@
+import * as Link from 'multiformats/link'
+import { DecodeFailure, EncodeFailure, asDID, asDIDWeb } from './lib.js'
+
+/**
+ * @typedef {import('../lib/api').Consumer} Consumer
+ * @typedef {import('../types').InferStoreRecord<Consumer>} ConsumerStoreRecord
+ * @typedef {import('../types').StoreRecord} StoreRecord
+ * @typedef {import('../lib/api').ConsumerKey} ConsumerKey
+ * @typedef {import('../types').InferStoreRecord<ConsumerKey>} ConsumerKeyStoreRecord
+ * @typedef {import('../lib/api').ConsumerListKey} ConsumerListKey
+ * @typedef {import('../types').InferStoreRecord<ConsumerListKey>} ConsumerListKeyStoreRecord
+ * @typedef {Pick<Consumer, 'consumer'|'provider'|'subscription'>} ConsumerList
+ */
+
+/** @type {import('../lib/api').Encoder<Consumer, ConsumerStoreRecord>} */
+export const encode = input => {
+  try {
+    return {
+      ok: {
+        consumer: input.consumer,
+        provider: input.provider,
+        subscription: input.subscription,
+        cause: input.cause.toString(),
```

Comment:
Well yes this is kind of what I was thinking with the UCAN invocation stream but as you rightly pointed out dag-json serialization of DIDs wouldn't have worked. I think since we have precedence for strings everywhere else here I'm inclined to keep as is.